pkg/sync/adapter.go

```diff
@@ -235,6 +265,65 @@ func getSyncFolder(projects []versionsCommon.DevfileProject) (string, error) {
 
 }
 
+// updateIndexWithWatchChanges uses the pushParameters.WatchDeletedFiles and pushParamters.WatchFiles to update
+// the existing index file; the index file is required to exist when this function is called.
+func updateIndexWithWatchChanges(pushParameters common.PushParameters) error {
```

Comment:
Could you add unit tests for this function if possible?

Reply:
Done.